wear/src/main/kotlin/au/com/shiftyjelly/pocketcasts/wear/MainActivity.kt

```diff
@@ -59,12 +66,41 @@ fun WearApp(themeType: Theme.ThemeType) {
                 WatchListScreen(navController::navigate, it.scrollableState)
             }
 
-            composable(NowPlayingScreen.route) { viewModel ->
-                viewModel.timeTextMode = NavScaffoldViewModel.TimeTextMode.Off
+            composable(NowPlayingScreen.route) {
+                it.timeTextMode = NavScaffoldViewModel.TimeTextMode.Off
+
+                // Listen for results from streaming confirmation screen
+                navController.currentBackStackEntry?.savedStateHandle
+                    ?.getStateFlow<StreamingConfirmationScreen.Result?>(StreamingConfirmationScreen.resultKey, null)
+                    ?.collectAsStateWithLifecycle()?.value?.let { streamingConfirmationResult ->
+                        val viewModel = hiltViewModel<NowPlayingViewModel>()
+                        LaunchedEffect(streamingConfirmationResult) {
+                            viewModel.onStreamingConfirmationResult(streamingConfirmationResult)
+                            // Clear result once consumed
+                            navController.currentBackStackEntry?.savedStateHandle
+                                ?.remove<StreamingConfirmationScreen.Result?>(StreamingConfirmationScreen.resultKey)
+                        }
+                    }
+
                 NowPlayingScreen(
                     navigateToEpisode = { episodeUuid ->
                         navController.navigate(EpisodeScreenFlow.navigateRoute(episodeUuid))
                     },
+                    showStreamingConfirmation = { navController.navigate(StreamingConfirmationScreen.route) },
+                )
+            }
+
+            composable(StreamingConfirmationScreen.route) {
+                it.timeTextMode = NavScaffoldViewModel.TimeTextMode.Off
+
+                StreamingConfirmationScreen(
+                    onFinished = { result ->
+                        navController.previousBackStackEntry?.savedStateHandle?.set(
+                            StreamingConfirmationScreen.resultKey,
+                            result
+                        )
```

Comment:
This is passing the result of what the user selected back to the previous screen.